runtimebp/cpu.go

```diff
@@ -0,0 +1,90 @@
+package runtimebp
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+// NumCPU returns the number of CPUs assigned to this running container.
+//
+// This is the container aware version of runtime.NumCPU.
+// It reads from the cgroup sysfs values.
+//
+// If the current process is not running inside a container,
+// or for whatever reason we failed to read the cgroup sysfs values,
+// it will fallback to runtime.NumCPU() instead.
+//
+// When fallback happens, it also prints the reason to stderr.
+func NumCPU() (n float64) {
+	const (
+		quotaPath  = "/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+		periodPath = "/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+	)
+
+	var err error
+	defer func() {
+		if err != nil {
+			// Fallback and log to stderr.
+			n = float64(runtime.NumCPU())
+			fmt.Fprintf(
+				os.Stderr,
+				"NumCPU: falling back to use runtime.NumCPU(): %v\n",
+				err,
+			)
+		}
+	}()
+
+	// Big enough buffer to read the number in the file wholly into memory.
+	buf := make([]byte, 1024)
+	var quota, period float64
+
+	quota, err = readNumberFromFile(quotaPath, buf)
+	if err != nil {
+		return
+	}
+
+	period, err = readNumberFromFile(periodPath, buf)
```

Comment:
Divide by zero in go will give you NaN instead of panic (https://play.golang.com/p/kl3nzAypb_M). `GOMAXPROCS` below will bound it inside [min, max] so we should be fine. But I'm gonna add a test for NaN.